mycpp/marksweep_heap.cc

```diff
@@ -0,0 +1,127 @@
+#include "mycpp/runtime.h"
+
+void MarkSweepHeap::Init(int collection_thresh) {
+  this->collection_thresh_ = collection_thresh;
+}
+
+void* MarkSweepHeap::Allocate(int byte_count) {
+  void* Result = calloc(byte_count, 1);
+  assert(Result);
+
+  this->all_allocations_.insert(Result);
+
+  this->current_heap_bytes_ += byte_count;
+  if (this->current_heap_bytes_ > this->collection_thresh_) {
+    Collect();
+  }
+
+  if (this->current_heap_bytes_ > this->collection_thresh_) {
```

Comment:
Make a note that current_heap_bytes_ is mutated hence the duplicate check

Also can we update current_heap_bytes_ ?